plugins/woocommerce-admin/client/customize-store/assembler-hub/iframe.jsx

```diff
@@ -24,75 +28,258 @@ import { useSelect } from '@wordpress/data';
-
-	const { resolvedAssets } = useSelect( ( select ) => {
-		const settings = select( blockEditorStore ).getSettings();
-
+	const { resolvedAssets, isPreviewMode } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const settings = getSettings();
 		return {
 			resolvedAssets: settings.__unstableResolvedAssets,
+			isPreviewMode: settings.__unstableIsPreviewMode,
 		};
 	}, [] );
-	const { styles = '', scripts = '' } = resolvedAssets;
 
+	const { styles = '', scripts = '' } = resolvedAssets;
+	const [ iframeDocument, setIframeDocument ] = useState();
+	const prevContainerWidth = useRef( 0 );
+	const [ bodyClasses, setBodyClasses ] = useState( [] );
 	const [ contentResizeListener, { height: contentHeight } ] =
 		useResizeObserver();
+	const [ containerResizeListener, { width: containerWidth } ] =
+		useResizeObserver();
+
 	const setRef = useRefEffect( ( node ) => {
 		node._load = () => {
 			setIframeDocument( node.contentDocument );
 		};
+		let iFrameDocument;
+		// Prevent the default browser action for files dropped outside of dropzones.
+		function preventFileDropDefault( event ) {
+			event.preventDefault();
+		}
 		function onLoad() {
 			const { contentDocument, ownerDocument } = node;
 			const { documentElement } = contentDocument;
+			iFrameDocument = contentDocument;
 
 			documentElement.classList.add( 'block-editor-iframe__html' );
 
+			// Ideally ALL classes that are added through get_body_class should
+			// be added in the editor too, which we'll somehow have to get from
+			// the server in the future (which will run the PHP filters).
 			setBodyClasses(
-				Array.from( ownerDocument?.body.classList ).filter(
+				Array.from( ownerDocument.body.classList ).filter(
 					( name ) =>
 						name.startsWith( 'admin-color-' ) ||
 						name.startsWith( 'post-type-' ) ||
 						name === 'wp-embed-responsive'
 				)
 			);
+
+			contentDocument.dir = ownerDocument.dir;
+
+			for ( const compatStyle of getCompatibilityStyles() ) {
+				if ( contentDocument.getElementById( compatStyle.id ) ) {
+					continue;
+				}
+
+				contentDocument.head.appendChild(
+					compatStyle.cloneNode( true )
+				);
+
+				if ( ! isPreviewMode ) {
+					// eslint-disable-next-line no-console
+					console.warn(
+						`${ compatStyle.id } was added to the iframe incorrectly. Please use block.json or enqueue_block_assets to add styles to the iframe.`,
+						compatStyle
+					);
+				}
+			}
+
+			iFrameDocument.addEventListener(
+				'dragover',
+				preventFileDropDefault,
+				false
+			);
+			iFrameDocument.addEventListener(
+				'drop',
+				preventFileDropDefault,
+				false
+			);
 		}
 
 		node.addEventListener( 'load', onLoad );
 
 		return () => {
 			delete node._load;
 			node.removeEventListener( 'load', onLoad );
+			iFrameDocument?.removeEventListener(
+				'dragover',
+				preventFileDropDefault
+			);
+			iFrameDocument?.removeEventListener(
+				'drop',
+				preventFileDropDefault
+			);
```

Comment:
Same concern for this logic

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/50070/commits/1015fd65e48a0f23ddd9194152e98ece8d020895